Data race in ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:82

нашел tsan, [лог](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-whitelist-run/12897511712/ya-x86-64-tsan/try_3/artifacts/logs/ydb/tests/stress/simple_queue/tests/test-results/py3test/testing_out_stuff/test_workload.py.TestYdbWorkload.test_column.log)
```
E   WARNING: ThreadSanitizer: data race (pid=112601)
E   Write of size 8 at 0x7220001bf130 by thread T48:
E   #0 AddDataSize /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:90:17 (ydbd+0x243d2493) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #1 AddStepDataSize /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:125:25 (ydbd+0x243d2493)
E   #2 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::FlushDuration() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:204:17 (ydbd+0x243d2493)
E   #3 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::Execute(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::IDataSource> const&) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:64:9 (ydbd+0x243d48ab) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #4 NKikimr::NOlap::NReader::NCommon::TStepAction::DoExecuteImpl() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:22:33 (ydbd+0x243d3502) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #5 NKikimr::NOlap::NReader::IDataTasksProcessor::ITask::DoExecute(std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/columnshard/engines/reader/common/conveyor_task.cpp:8:19 (ydbd+0x24453bcd) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #6 NKikimr::NConveyor::ITask::Execute(std::__y1::shared_ptr<NKikimr::NConveyor::TTaskSignals>, std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/conveyor/usage/abstract.cpp:13:36 (ydbd+0x2439ebd6) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #7 NKikimr::NConveyor::TWorker::ExecuteTask(NKikimr::NConveyor::TWorkerTask const&) /-S/ydb/core/tx/conveyor/service/worker.cpp:10:36 (ydbd+0x303a8412) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #8 NKikimr::NConveyor::TWorker::HandleMain(TAutoPtr<NActors::TEventHandle<NKikimr::NConveyor::TEvInternal::TEvNewTask>, TDelete>&) /-S/ydb/core/tx/conveyor/service/worker.cpp:28:9 (ydbd+0x303a894e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #9 NKikimr::NConveyor::TWorker::StateMain(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/core/tx/conveyor/service/worker.h:93:13 (ydbd+0x303a4964) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #10 NActors::TActorCallbackBehaviour::Receive(NActors::IActor*, TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:260:9 (ydbd+0x17a00fd3) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #11 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.h:548:23 (ydbd+0x17a5bb62) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #12 NActors::TGenericExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:281:28 (ydbd+0x17a57bcd) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #13 NActors::TGenericExecutorThread::ProcessExecutorPool(NActors::IExecutorPool*)::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:475:39 (ydbd+0x17a5da0e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #14 NActors::TGenericExecutorThread::ProcessExecutorPool(NActors::IExecutorPool*) /-S/ydb/library/actors/core/executor_thread.cpp:529:13 (ydbd+0x17a5d0de) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #15 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:560:9 (ydbd+0x17a5e799) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #16 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:383:45 (ydbd+0x1666eec2) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #17 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:244:20 (ydbd+0x1666f39c) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   
E   Previous write of size 8 at 0x7220001bf130 by thread T47:
E   #0 AddDataSize /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:90:17 (ydbd+0x243d2493) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #1 AddStepDataSize /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:125:25 (ydbd+0x243d2493)
E   #2 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::FlushDuration() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:204:17 (ydbd+0x243d2493)
E   #3 NKikimr::NOlap::NReader::NCommon::TFetchingScriptCursor::Execute(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::IDataSource> const&) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:64:9 (ydbd+0x243d48ab) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #4 NKikimr::NOlap::NReader::NCommon::TStepAction::DoExecuteImpl() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:22:33 (ydbd+0x243d3502) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #5 NKikimr::NOlap::NReader::IDataTasksProcessor::ITask::DoExecute(std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/columnshard/engines/reader/common/conveyor_task.cpp:8:19 (ydbd+0x24453bcd) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #6 NKikimr::NConveyor::ITask::Execute(std::__y1::shared_ptr<NKikimr::NConveyor::TTaskSignals>, std::__y1::shared_ptr<NKikimr::NConveyor::ITask> const&) /-S/ydb/core/tx/conveyor/usage/abstract.cpp:13:36 (ydbd+0x2439ebd6) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #7 NKikimr::NConveyor::TWorker::ExecuteTask(NKikimr::NConveyor::TWorkerTask const&) /-S/ydb/core/tx/conveyor/service/worker.cpp:10:36 (ydbd+0x303a8412) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #8 NKikimr::NConveyor::TWorker::HandleMain(TAutoPtr<NActors::TEventHandle<NKikimr::NConveyor::TEvInternal::TEvNewTask>, TDelete>&) /-S/ydb/core/tx/conveyor/service/worker.cpp:28:9 (ydbd+0x303a894e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #9 NKikimr::NConveyor::TWorker::StateMain(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/core/tx/conveyor/service/worker.h:93:13 (ydbd+0x303a4964) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #10 NActors::TActorCallbackBehaviour::Receive(NActors::IActor*, TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:260:9 (ydbd+0x17a00fd3) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #11 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.h:548:23 (ydbd+0x17a5bb62) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #12 NActors::TGenericExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:281:28 (ydbd+0x17a57bcd) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #13 NActors::TGenericExecutorThread::ProcessExecutorPool(NActors::IExecutorPool*)::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:475:39 (ydbd+0x17a5da0e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #14 NActors::TGenericExecutorThread::ProcessExecutorPool(NActors::IExecutorPool*) /-S/ydb/library/actors/core/executor_thread.cpp:529:13 (ydbd+0x17a5d0de) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #15 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:560:9 (ydbd+0x17a5e799) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #16 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:383:45 (ydbd+0x1666eec2) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #17 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:244:20 (ydbd+0x1666f39c) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   
E   Location is heap block of size 128 at 0x7220001bf100 allocated by thread T46:
E   #0 operator new(unsigned long) /-S/contrib/libs/clang18-rt/lib/tsan/rtl/tsan_new_delete.cpp:64:3 (ydbd+0x164a8117) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #1 __libcpp_operator_new<unsigned long> /-S/contrib/libs/cxxsupp/libcxx/include/new:271:10 (ydbd+0x243d9290) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #2 __libcpp_allocate /-S/contrib/libs/cxxsupp/libcxx/include/new:295:10 (ydbd+0x243d9290)
E   #3 allocate /-S/contrib/libs/cxxsupp/libcxx/include/__memory/allocator.h:125:32 (ydbd+0x243d9290)
E   #4 allocate /-S/contrib/libs/cxxsupp/libcxx/include/__memory/allocator_traits.h:281:16 (ydbd+0x243d9290)
E   #5 __allocation_guard<std::__y1::allocator<NKikimr::NOlap::NReader::NCommon::TAssemblerStep> > /-S/contrib/libs/cxxsupp/libcxx/include/__memory/allocation_guard.h:56:16 (ydbd+0x243d9290)
E   #6 allocate_shared<NKikimr::NOlap::NReader::NCommon::TAssemblerStep, std::__y1::allocator<NKikimr::NOlap::NReader::NCommon::TAssemblerStep>, std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::TColumnsSet> &, TBasicString<char, std::__y1::char_traits<char> > &, 0> /-S/contrib/libs/cxxsupp/libcxx/include/__memory/shared_ptr.h:857:46 (ydbd+0x243d9290)
E   #7 make_shared<NKikimr::NOlap::NReader::NCommon::TAssemblerStep, std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::TColumnsSet> &, TBasicString<char, std::__y1::char_traits<char> > &, 0> /-S/contrib/libs/cxxsupp/libcxx/include/__memory/shared_ptr.h:866:10 (ydbd+0x243d9290)
E   #8 std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::TAssemblerStep> NKikimr::NOlap::NReader::NCommon::TFetchingScript::AddStep<NKikimr::NOlap::NReader::NCommon::TAssemblerStep, std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::TColumnsSet>, TBasicString<char, std::__y1::char_traits<char>>>(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::TColumnsSet>, TBasicString<char, std::__y1::char_traits<char>>) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:146:23 (ydbd+0x243d9290)
E   #9 NKikimr::NOlap::NReader::NCommon::TColumnsAccumulator::AddAssembleStep(NKikimr::NOlap::NReader::NCommon::TFetchingScript&, NKikimr::NOlap::NReader::NCommon::TColumnsSetIds const&, TBasicString<char, std::__y1::char_traits<char>> const&, NKikimr::NOlap::NReader::NCommon::EStageFeaturesIndexes, bool) /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp:161:16 (ydbd+0x243d8a5e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #10 NKikimr::NOlap::NReader::NPlain::TSpecialReadContext::BuildColumnsFetchingPlan(bool, bool, bool, bool, bool, bool) const /-S/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/context.cpp:127:21 (ydbd+0x243ff135) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #11 NKikimr::NOlap::NReader::NPlain::TSpecialReadContext::DoGetColumnsFetchingPlan(std::__y1::shared_ptr<NKikimr::NOlap::NReader::NCommon::IDataSource> const&) /-S/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/context.cpp:69:26 (ydbd+0x243fb67e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #12 GetColumnsFetchingPlan /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/context.h:50:16 (ydbd+0x243ad499) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #13 GetColumnsFetchingPlan<NKikimr::NOlap::NReader::NPlain::IDataSource> /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/context.h:46:16 (ydbd+0x243ad499)
E   #14 NKikimr::NOlap::NReader::NPlain::TScanHead::Start() /-S/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/scanner.cpp:91:42 (ydbd+0x243ad499)
E   #15 NKikimr::NOlap::NReader::NPlain::TPlainReadData::DoStart() /-S/ydb/core/tx/columnshard/engines/reader/plain_reader/iterator/plain_read_data.h:22:25 (ydbd+0x243a393a) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #16 NKikimr::NOlap::NReader::IDataReader::Start() /-S/ydb/core/tx/columnshard/engines/reader/abstract/read_context.h:172:16 (ydbd+0x2439ddac) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #17 NKikimr::NOlap::NReader::NCommon::TColumnShardScanIterator::Start() /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/iterator.h:74:29 (ydbd+0x2439a325) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #18 NKikimr::NOlap::NReader::TColumnShardScan::Bootstrap(NActors::TActorContext const&) /-S/ydb/core/tx/columnshard/engines/reader/actor/actor.cpp:88:38 (ydbd+0x24607167) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #19 NActors::TActorBootstrapped<NKikimr::NOlap::NReader::TColumnShardScan>::StateBootstrap(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor_bootstrapped.h:22:22 (ydbd+0x24616f9a) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #20 NActors::TActorCallbackBehaviour::Receive(NActors::IActor*, TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:260:9 (ydbd+0x17a00fd3) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #21 NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.h:548:23 (ydbd+0x17a5bb62) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #22 NActors::TGenericExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:281:28 (ydbd+0x17a57bcd) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #23 NActors::TGenericExecutorThread::ProcessExecutorPool(NActors::IExecutorPool*)::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:475:39 (ydbd+0x17a5da0e) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #24 NActors::TGenericExecutorThread::ProcessExecutorPool(NActors::IExecutorPool*) /-S/ydb/library/actors/core/executor_thread.cpp:529:13 (ydbd+0x17a5d0de) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #25 NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:560:9 (ydbd+0x17a5e799) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #26 void* (anonymous namespace)::ThreadProcWrapper<ISimpleThread>(void*) /-S/util/system/thread.cpp:383:45 (ydbd+0x1666eec2) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #27 (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:244:20 (ydbd+0x1666f39c) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   
E   Thread T48 'ydbd.User' (tid=114107, running) created by main thread at:
E   #0 pthread_create /-S/contrib/libs/clang18-rt/lib/tsan/rtl/tsan_interceptors_posix.cpp:1023:3 (ydbd+0x16427ebb) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #1 Start /-S/util/system/thread.cpp:229:27 (ydbd+0x1666e4c9) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #2 TThread::Start() /-S/util/system/thread.cpp:314:34 (ydbd+0x1666e4c9)
E   #3 NActors::TBasicExecutorPool::Start() /-S/ydb/library/actors/core/executor_pool_basic.cpp:475:32 (ydbd+0x17a43488) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #4 NActors::TCpuManager::Start() /-S/ydb/library/actors/core/cpu_manager.cpp:82:32 (ydbd+0x17a13b83) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #5 NActors::TActorSystem::Start() /-S/ydb/library/actors/core/actorsystem.cpp:312:21 (ydbd+0x17a0d72a) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #6 NKikimr::TKikimrRunner::KikimrStart() /-S/ydb/core/driver_lib/run/run.cpp:1744:22 (ydbd+0x27f30dd1) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #7 NKikimr::MainRun(NKikimr::TKikimrRunConfig const&, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:46:17 (ydbd+0x269f2c7d) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #8 NKikimr::NDriverClient::TClientCommandServer::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_server.cpp:51:12 (ydbd+0x27eb15ac) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #9 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:258:12 (ydbd+0x26a2cb91) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #10 NYdb::NConsoleClient::TClientCommandTree::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:417:33 (ydbd+0x26a2f71b) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #11 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:258:12 (ydbd+0x26a2cb91) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #12 NYdb::NConsoleClient::TClientCommand::Process(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:235:12 (ydbd+0x26a2c2ed) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #13 NKikimr::NDriverClient::NewClient(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_root.cpp:72:26 (ydbd+0x26a25754) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #14 NKikimr::Main(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:148:20 (ydbd+0x269f517b) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #15 ParameterizedMain(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:194:16 (ydbd+0x269f64d3) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #16 main /-S/ydb/apps/ydbd/main.cpp:24:12 (ydbd+0x16403843) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   
E   Thread T47 'ydbd.User' (tid=114100, running) created by main thread at:
E   #0 pthread_create /-S/contrib/libs/clang18-rt/lib/tsan/rtl/tsan_interceptors_posix.cpp:1023:3 (ydbd+0x16427ebb) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #1 Start /-S/util/system/thread.cpp:229:27 (ydbd+0x1666e4c9) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #2 TThread::Start() /-S/util/system/thread.cpp:314:34 (ydbd+0x1666e4c9)
E   #3 NActors::TBasicExecutorPool::Start() /-S/ydb/library/actors/core/executor_pool_basic.cpp:475:32 (ydbd+0x17a43488) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #4 NActors::TCpuManager::Start() /-S/ydb/library/actors/core/cpu_manager.cpp:82:32 (ydbd+0x17a13b83) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #5 NActors::TActorSystem::Start() /-S/ydb/library/actors/core/actorsystem.cpp:312:21 (ydbd+0x17a0d72a) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #6 NKikimr::TKikimrRunner::KikimrStart() /-S/ydb/core/driver_lib/run/run.cpp:1744:22 (ydbd+0x27f30dd1) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #7 NKikimr::MainRun(NKikimr::TKikimrRunConfig const&, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:46:17 (ydbd+0x269f2c7d) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #8 NKikimr::NDriverClient::TClientCommandServer::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_server.cpp:51:12 (ydbd+0x27eb15ac) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #9 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:258:12 (ydbd+0x26a2cb91) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #10 NYdb::NConsoleClient::TClientCommandTree::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:417:33 (ydbd+0x26a2f71b) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #11 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:258:12 (ydbd+0x26a2cb91) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #12 NYdb::NConsoleClient::TClientCommand::Process(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:235:12 (ydbd+0x26a2c2ed) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #13 NKikimr::NDriverClient::NewClient(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_root.cpp:72:26 (ydbd+0x26a25754) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #14 NKikimr::Main(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:148:20 (ydbd+0x269f517b) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #15 ParameterizedMain(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:194:16 (ydbd+0x269f64d3) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #16 main /-S/ydb/apps/ydbd/main.cpp:24:12 (ydbd+0x16403843) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   
E   Thread T46 'ydbd.User' (tid=114087, running) created by main thread at:
E   #0 pthread_create /-S/contrib/libs/clang18-rt/lib/tsan/rtl/tsan_interceptors_posix.cpp:1023:3 (ydbd+0x16427ebb) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #1 Start /-S/util/system/thread.cpp:229:27 (ydbd+0x1666e4c9) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #2 TThread::Start() /-S/util/system/thread.cpp:314:34 (ydbd+0x1666e4c9)
E   #3 NActors::TBasicExecutorPool::Start() /-S/ydb/library/actors/core/executor_pool_basic.cpp:475:32 (ydbd+0x17a43488) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #4 NActors::TCpuManager::Start() /-S/ydb/library/actors/core/cpu_manager.cpp:82:32 (ydbd+0x17a13b83) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #5 NActors::TActorSystem::Start() /-S/ydb/library/actors/core/actorsystem.cpp:312:21 (ydbd+0x17a0d72a) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #6 NKikimr::TKikimrRunner::KikimrStart() /-S/ydb/core/driver_lib/run/run.cpp:1744:22 (ydbd+0x27f30dd1) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #7 NKikimr::MainRun(NKikimr::TKikimrRunConfig const&, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:46:17 (ydbd+0x269f2c7d) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #8 NKikimr::NDriverClient::TClientCommandServer::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_server.cpp:51:12 (ydbd+0x27eb15ac) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #9 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:258:12 (ydbd+0x26a2cb91) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #10 NYdb::NConsoleClient::TClientCommandTree::Run(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:417:33 (ydbd+0x26a2f71b) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #11 NYdb::NConsoleClient::TClientCommand::ValidateAndRun(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:258:12 (ydbd+0x26a2cb91) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #12 NYdb::NConsoleClient::TClientCommand::Process(NYdb::NConsoleClient::TClientCommand::TConfig&) /-S/ydb/public/lib/ydb_cli/common/command.cpp:235:12 (ydbd+0x26a2c2ed) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #13 NKikimr::NDriverClient::NewClient(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/cli_utils/cli_cmds_root.cpp:72:26 (ydbd+0x26a25754) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #14 NKikimr::Main(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:148:20 (ydbd+0x269f517b) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #15 ParameterizedMain(int, char**, std::__y1::shared_ptr<NKikimr::TModuleFactories>) /-S/ydb/core/driver_lib/run/main.cpp:194:16 (ydbd+0x269f64d3) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   #16 main /-S/ydb/apps/ydbd/main.cpp:24:12 (ydbd+0x16403843) (BuildId: 1c7a4e86a31b496c38e38bd1c09219cc5dda4b66)
E   
E   SUMMARY: ThreadSanitizer: data race /-S/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h:90:17 in AddDataSize```

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.cpp
@@ -121,7 +121,7 @@ void TFetchingScript::Allocation(const std::set<ui32>& entityIds, const EStageFe
 TString IFetchingStep::DebugString() const {
     TStringBuilder sb;
     sb << "name=" << Name << ";duration=" << GetSumDuration() << ";"
-       << "size=" << 1e-9 * SumSize << ";details={" << DoDebugString() << "};";
+       << "size=" << 1e-9 * GetSumSize() << ";details={" << DoDebugString() << "};";
     return sb;
 }
 

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
@@ -67,8 +67,8 @@ public:
 class IFetchingStep: public TNonCopyable {
 private:
     YDB_READONLY_DEF(TString, Name);
-    YDB_READONLY_DEF(TAtomicCounter, SumDurationUs);
-    YDB_READONLY_DEF(TAtomicCounter, SumSize);
+    TAtomicCounter SumDuration;
+    TAtomicCounter SumSize;
     TFetchingStepSignals Signals;
 
 protected:
@@ -82,8 +82,12 @@ public:
         return TDuration::MicroSeconds(SumDuration.Val());
     }
 
+    ui64 GetSumSize() const {
+        return SumSize.Val();
+    }
+
     void AddDuration(const TDuration d) {
-        SumDurationUs.Add(d.MicroSeconds());
+        SumDuration.Add(d.MicroSeconds());
         Signals.AddDuration(d);
     }
     void AddDataSize(const ui64 size) {

--- a/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
+++ b/ydb/core/tx/columnshard/engines/reader/common_reader/iterator/fetching.h
@@ -67,8 +67,8 @@ public:
 class IFetchingStep: public TNonCopyable {
 private:
     YDB_READONLY_DEF(TString, Name);
-    TAtomicCounter SumDuration;
-    YDB_READONLY(ui64, SumSize, 0);
+    YDB_READONLY_DEF(TAtomicCounter, SumDurationUs);
+    YDB_READONLY_DEF(TAtomicCounter, SumSize);
     TFetchingStepSignals Signals;
 
 protected:
@@ -83,11 +83,11 @@ public:
     }
 
     void AddDuration(const TDuration d) {
-        SumDuration.Add(d.MicroSeconds());
+        SumDurationUs.Add(d.MicroSeconds());
         Signals.AddDuration(d);
     }
     void AddDataSize(const ui64 size) {
-        SumSize += size;
+        SumSize.Add(size);
         Signals.AddBytes(size);
     }
 

